src/tsconfig.json

```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../tsconfig.lib.json"
```

Comment:
Not sure if that's a good idea. We have two separate TS configs: one (`tsconfig.lib.json`) for the library which is respected when building it and the other one (`tsconfig.json`) which is both for the IDEs and for the Demo project. What's the reason of mixing them?

Reply:
This config helps IDE to locate the right config for the files in directory. Otherwise it will use tsconfig.json which still has strict mode set to `false` and IDE's TS service won't apply strict mode to the files so they would only be detected when running tsc commend and passing config. Hence why this file is needed.

Reply:
> Otherwise it will use tsconfig.json which still has strict mode set to false 

So why don't we set it to true there?

Reply:
Didn't turned on strict mode in `tsconfig.json` because there were to many TS errors in the demo with it turned on and dealing with them seemed not to be worthwhile. But after your comment I had another glance at those errors and they turned out not too cumbersome to fix. So now strict mode is enabled for all the code.